unicon_runner/executor/sandbox.py

```diff
@@ -29,27 +29,34 @@ def _cmd(self, cwd: Path) -> tuple[list[str], dict[str, str]]:
         # NOTE: `uv` binary is assumed to be stored under `~/.cargo/bin/`
         # We are using `uv` as the environment manager and program runner
         uv_path = Path("~/.cargo/bin/uv").expanduser()
-        # NOTE: We need to bind the uv cache folder to access uv-managed python executables
-        uv_cache_path = Path("~/.local/share/uv").expanduser()
+        # NOTE: We need to bind the uv app state folder to access uv-managed python executables
+        uv_app_state_path = Path("~/.local/share/uv").expanduser()
+        uv_cache_path = Path("~/.cache/uv").expanduser()
 
+        # fmt: off
         return [
             CONTY_PATH,
-            "--bind",
-            str(cwd.absolute()),
-            str(cwd),
-            "--ro-bind",
-            *([str(uv_path)] * 2),
-            "--ro-bind",
-            *([str(uv_cache_path)] * 2),
+            "--ro-bind", *(["/"] * 2),
+            "--ro-bind", *([str(uv_path)] * 2),
+            "--ro-bind", *([str(uv_app_state_path)] * 2),
+            "--bind", *([str(uv_cache_path)] * 2),
+            # R/W bind to the root working directory
+            "--bind", *([str(cwd.parents[0])] * 2),
+            # NOTE: Mount `procfs` to allow access to process information
+            # This seems be required for GPU workloads
+            "--proc", "/proc",
+            # Bind /dev to allow access to devices
+            "--dev-bind", *(["/dev"] * 2),
             str(cwd / self.ENTRYPOINT),
         ], {
             # Conty specific environment variables
-            "SANDBOX": "1",
-            "SANDBOX_LEVEL": "1",
-            "QUIET_MODE": "1",
+            "SANDBOX": "1", "SANDBOX_LEVEL": "1", "QUIET_MODE": "1",
+            # NOTE: `conty` installs NVIDIA drivers in the sandbox, so we need to disable it
+            "NVIDIA_HANDLER": "-1",
             # NOTE: We need to unset VIRTUAL_ENV to prevent uv from using the wrong base python interpreter
             "VIRTUAL_ENV": "''",
         }
+        # fmt: off
 
     async def _collect(self, proc: asyncio.subprocess.Process) -> ExecutorResult:
         stdout, stderr = await proc.communicate()
```

Comment:
workspace from base.py may need to be passed here to fix the filepath (it is currently /temp/exit_code) - but even after correcting it to /tmp/whatever-long-code-here-uuid-thing/exit_code --> it can't find it (maybe its the slurm cleanup thing)